### PR TITLE
Set up language-specific formio translation endpoints

### LINKED ISSUE #855: Optimise passing Formio translation file from backend to frontend
Currently the frontend retrieves the JSON file from the backend with an API call to the backend (with this view: ).

This should be improved so that ngix serves the file

### PR DESCRIPTION
**Pull request by sergei-maertens**
Related to #2255 and fixes #855.

This sets up (new) endpoints to be used by the SDK to retrieve language-specific Formio translations. This also normalizes the URL structure for translation related endpoints and re-organizes the code.

### CODE REVIEW

File: src/openforms/translations/tests/test_views.py [hunk 1]
```diff
@@ -1,12 +1,46 @@
...
+    def test_language_code_param(self):
+        endpoint = reverse("api:i18n:formio-translations", kwargs={"language": "nl"})
+
+        response = self.client.get(endpoint)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response["Content-Type"], "application/json")
+        content = b"".join(x for x in response.streaming_content)
+        translations = json.loads(content)
+        self.assertNotIn("nl", translations)
```
[the remark this answers is not in the document]
Comment (CharString) [reply src/openforms/translations/tests/test_views.py:27]: Oh it looks like the response is a dict `{"default message": "translation"}`. Just drop the assertion then.

File: src/openforms/translations/api/views.py [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
+from openforms.api.drf_spectacular.functional import lazy_enum
 from openforms.api.serializers import ExceptionSerializer, ValidationErrorSerializer
 from openforms.translations.utils import set_language_cookie
 
 from .serializers import LanguageCodeSerializer, LanguageInfoSerializer
 
 
+def get_language_codes() -> List[str]:
+    return [language[0] for language in settings.LANGUAGES]
+
```
Comment (CharString) [inline src/openforms/translations/api/views.py:30]: Maybe reuse this in the other view that references this same list `LanguageInfoView.get()`.

**Review by CharString — CHANGES_REQUESTED** [review]

File: src/openforms/translations/tests/test_views.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/translations/tests/test_views.py:27]: the deprecated endpoint would return a document `{"nl": {"key": "value"}}`, while the new ones return just `{"key": "value"}`. I decided to let the SDK deal with that (Formio expects that `nl` top level key`) so we can efficiently use `FileResponse` and not have to read a JSON file, load it into a python dict only to have DRF serialize it back to JSON again.

The assertion makes explicit that the client needs to do some extra work and that this is intentional :)

File: src/openforms/translations/api/views.py [hunk 3]
```diff
@@ -88,3 +99,46 @@ def put(self, request: Request) -> Response:
...
+    summary=_("Get FormIO translations"),
+    description=_("Retrieve the translations for the strings used by FormIO."),
+    tags=["translations"],
+    parameters=[
+        OpenApiParameter(
+            name="language",
+            location=OpenApiParameter.PATH,
+            description=_("Language code to retrieve the messages for."),
+            type=str,
+            enum=lazy_enum(get_language_codes),
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/translations/api/views.py:114]: I'm not sure I can reference to an enum that is yet-to-be-generated by drf-spectacular at this level. I'll take a look to see what can be done - the duplication bothers me too. Maybe the lazy enum can be fed to DRF choices field (but then we lose the nice key/label display :( )

[the remark this answers is not in the document]
Comment (CharString) [reply src/openforms/translations/api/views.py:114]: I think it collects all enum like things and checks if any look the same. I remember I defined the AvailableLanaguagesEnum in the settings.py.. Does it barf if we leave off the lazy_enum wrapper?

[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/translations/api/views.py:114]: will leave that for next week to check, but in general evaluating django settings at import time is frowned upon as it causes problems in tests with `@override_settings`

**Review by SilviaAmAm — APPROVED** [review]

File: src/openforms/translations/api/views.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/translations/api/views.py:110]: I've looked into it - drf-spectacular only runs this post processor hook on the spec's `components.schemas.*`, while the operation parameters are outputted under the spec's `paths.*` and thus do not end up there.

using or not using `lazy_enum` does not make any difference, so I'm leaving this as is.

File: src/openforms/translations/api/views.py (hunk 2 above)
Comment (sergei-maertens) [reply src/openforms/translations/api/views.py:26]: done that!

**Pull request merged**